snmplib/snmpusm.c

```diff
@@ -1427,11 +1429,16 @@ usm_generate_out_msg(int msgProcModel,  /* (UNUSED) */
                       * * USM if needed.
                       */
                      size_t scopedPduLen,       /* IN - scopedPdu length. */
-                     const void *secStateRef, /* IN */
+                     const void *secStateRef,   /* IN */
                      /*
                       * secStateRef, pointer to cached info provided only for
                       * * Response, otherwise NULL.
                       */
+                     const u_long sessFlags,    /* IN */
```

Comment:
Why is this new argument needed? Isn't it sufficient to check whether or not the new sessUser argument is not NULL?

Reply:
Good point !
Done